crypto/src/main/java/net/corda/v5/crypto/CompositeKeyNodeAndWeight.java

```diff
@@ -1,29 +1,60 @@
 package net.corda.v5.crypto;
+
 import org.jetbrains.annotations.NotNull;
 
 import java.security.PublicKey;
 
 
 /**
  * A simple data class for passing keys and weights into CompositeKeyGenerator
- *
- * @param node A public key
- * @param weight The weight for that key, must be greater than zero.
  */
 public final class CompositeKeyNodeAndWeight {
+    /**
+     * Construct a key
+     *
+     * @param node   A public key
+     * @param weight The weight for that key, must be greater than zero.
+     */
+    public CompositeKeyNodeAndWeight(@NotNull PublicKey node, int weight) {
+        super();
+        this.node = node;
+        this.weight = weight;
+        if (this.weight <= 0)
```

Comment:
Let's check parameter instead and move check prior to populating properties

Reply:
done

Reply:
What happened is that I started out by disassembling the bytecode, and Kotlin often initialises arguments before constructor logic.